app.py

```diff
@@ -32,6 +31,45 @@ def wrapper(*args, **kw):
     return wrapper
 
 
+def recover_identity(f):
+    @wraps(f)
+    def wrapper(*args, **kw):
+        # Authorization request header format:
+        # Authorization: Signature <signature_base64_encoded>
+        authorization = request.headers.get('Authorization')
+        if not authorization:
+            return jsonify(error='missing Authorization in request header'), 401
+
+        if authorization.count(' ') != 1:
```

Comment:
We don't need to use `.count` here, `.split` itself is enough:
```python
authorization_parts = authorization.split(' ')
if len(authorization_parts) != 2:
  ...
authentication_type, signature_base64_encoded = authorization_parts
```

Reply:
optimized